docs/user/04-metrics.md

```diff
@@ -1,3 +1,395 @@
+
+The agent is configured with a generic scrape configuration, which uses annotations to specify the endpoints to scrape in the cluster. 
+Having the annotations in place is everything you need for metrics ingestion to start automatically.
+
+Put the following annotations either to a service that resolves your metrics port, or directly to the pod:
+
+```yaml
+prometheus.io/scrape: "true"   # mandatory to enable automatic scraping
+prometheus.io/scheme: https    # optional, default is "http" if no Istio sidecar is used. When using a sidecar (Pod has label `security.istio.io/tlsMode=istio`), the default is "https". Use "https" to scrape workloads using Istio client certificates.
+prometheus.io/port: "1234"     # optional, configure the port under which the metrics are exposed
+prometheus.io/path: /myMetrics # optional, configure the path under which the metrics are exposed
+```
+
+> **NOTE:** The agent can scrape endpoints even if the workload uses Istio and accepts only mTLS communication.
+
+### Step 7: Activate runtime metrics
+To enable collection of runtime metrics for your pods, define a MetricPipeline having runtime enabled as input:
+```yaml
+apiVersion: telemetry.kyma-project.io/v1alpha1
+kind: MetricPipeline
+metadata:
+name: backend
+spec:
+  input:
+    application:
+      runtime:
+        enabled: true
+  output:
+    otlp:
+      endpoint:
+        value: https://backend.example.com:4317
+```
+
+### Step 8: Activate Istio metrics
+To enable collection of istio metrics for your pods, define a MetricPipeline having istio enabled as input:
+```yaml
+apiVersion: telemetry.kyma-project.io/v1alpha1
+kind: MetricPipeline
+metadata:
+name: backend
+spec:
+  input:
+    application:
+      istio:
+        enabled: true
+  output:
+    otlp:
+      endpoint:
+        value: https://backend.example.com:4317
+```
+
+## Limitations
+
+The metric gateway setup is designed using the following assumptions:
+- The collector has no autoscaling options yet and has a limited resource setup of 1 CPU and 1 GiB memory.
+- Batching is enabled, and a batch will contain up to 512 metrics/batch.
+- An unavailability of a destination must be survived for 5 minutes without direct loss of metric data.
+- An average metric consists of 40 attributes with 64 character length.
+
+This leads to the following limitations:
+### Throughput
+The maximum throughput is 4200 metric/sec ~= 15.000.000 metrics/hour. If more data must be ingested, it can result in a refusal of more data.
+
+### Unavailability of output
+For up to 5 minutes, a retry for data is attempted when the destination is unavailable. After that, data is dropped.
+
+### No guaranteed delivery
+The used buffers are volatile. If the gateway or agent instances crash, metric data can be lost.
+
+### Multiple MetricPipeline support
+
+Up to three MetricPipeline resources at a time are supported at the moment.
```

Comment:
```suggestion
Up to three MetricPipeline resources at a time are supported.
```

Once we support more pipelines, we announce it as a feature and describe it here. No need to mention it yet.